lib/agenda/define.ts

```diff
@@ -12,18 +27,18 @@ const debug = createDebugger('agenda:define');
  * @param options options for job to run
  * @param processor function to be called to run actual job
  */
-export const define = function(this: Agenda, name: string, options: any, processor: () => void) {
-  if (!processor) {
-    processor = options;
+export const define = function(this: Agenda, name: string, options: DefineOptions | (() => void), processor?: () => void) {
```

Comment:
processor is not exactly () => void it has two valid signatures:
`processor: ((job: Job) => Promise<void>) | ((job: Job, done) => void),`

I would suggest you make a `type Processor = processor: ((job: Job) => Promise<void>) | ((job: Job, done) => void);` and use this here twice.

Reply:
`done` should be typed.